core/transaction_pool.go

```diff
@@ -100,6 +112,10 @@ func (pool *TxPool) eventLoop() {
 			pool.resetState()
 		case GasPriceChanged:
 			pool.minGasPrice = ev.Price
+		case RemovedTransactionEvent:
+			pool.mu.Unlock()
+			pool.AddTransactions(ev.Txs)
```

Comment:
There's a small logical omission here that might have implications.  `AddTransactions` doesn't keep track of transaction ownership (https://github.com/ethereum/go-ethereum/pull/1669/files#diff-d37f7abd23589257d36b8fa71ac686faR331). I.e. even if the transaction was originally owned by this node, that meta info is lost.

However, given that the transaction was already integrated once, we might assume that the network is aware and a miner will reintegrate it. So it's not such a big issue, but in than scenario maybe we should add a comment here or in `AddTransactions` to highlight tht this scenario was thought of and why it works ok.